File: android/app/src/main/java/com/soobinpark/appcraft/readingbible/domain/repository/BibleRepository.kt
package com.soobinpark.appcraft.readingbible.domain.repository

import com.soobinpark.appcraft.readingbible.domain.model.BibleBook
import com.soobinpark.appcraft.readingbible.domain.model.BibleVerse
import com.soobinpark.appcraft.readingbible.domain.model.BibleVersion
import java.io.File

interface BibleRepository {
    suspend fun scanVersions(root: File): List<BibleVersion>

    suspend fun readChapter(
        version: BibleVersion,
        book: BibleBook,
        chapter: Int,
    ): List<BibleVerse>
}

File: ios/Sources/Domain/Repositories/BibleRepository.swift
import Foundation

protocol BibleRepository: Sendable {
    func scanVersions(at root: URL) async throws -> [BibleVersion]
    func readChapter(version: BibleVersion, book: BibleBook, chapter: Int) async throws -> [BibleVerse]
    func warmUpVersion(_ version: BibleVersion, onProgress: @Sendable (Int, Int, String) -> Void) async throws
    func search(version: BibleVersion, query: String, onProgress: @Sendable (Int, Int, String) -> Void) async throws -> [BibleSearchResult]
    func diagnose(at root: URL) async throws -> BibleFileDiagnostic
}
